stdlib/runtime/gc.gr

```diff
@@ -28,15 +28,102 @@ import WasmI32, {
   eq as (==)
 } from "runtime/unsafe/wasmi32"
 
+// Imported directly here to avoid cyclic dependency
+import foreign wasm fd_write : (WasmI32, WasmI32, WasmI32, WasmI32) -> WasmI32 from "wasi_snapshot_preview1"
+import foreign wasm fd_sync : (WasmI32) -> WasmI32 from "wasi_snapshot_preview1"
+
 primitive (&&) : (Bool, Bool) -> Bool = "@and"
 primitive (||) : (Bool, Bool) -> Bool = "@or"
 primitive throw : Exception -> a = "@throw"
 primitive ignore : a -> Void = "@ignore"
+primitive box : a -> Box<a> = "@box"
+primitive unbox : Box<a> -> a = "@unbox"
 
 exception DecRefError
 
+export let decimalCount32 = box((n) => 0n)
+export let utoa32Buf = box((a, b, c) => void)
+
+let mut _DEBUG = false
+
 let _HEADER_SIZE = 8n
 
+// incRef debug messages: "decRef: 0xNNNNNN (prev count: nn)\n"
+let _INCREF_DEBUG_STR_1 = "incRef: 0x"
+let _INCREF_DEBUG_STR_2 = " (prev count: "
+let _INCREF_DEBUG_STR_3 = "; tag: "
+let _INCREF_DEBUG_STR_4 = ")\n"
+
+// decRef debug messages: "decRef: 0xNNNNNN (prev count: nn)\n"
+let _DECREF_DEBUG_STR_1 = "decRef: 0x"
+let _DECREF_DEBUG_STR_2 = " (prev count: "
+let _DECREF_DEBUG_STR_3 = "; ignoreZeros: true)\n"
+let _DECREF_DEBUG_STR_4 = "; ignoreZeros: false)\n"
+
+let logIncRef = (userPtr: WasmI32, refCount: WasmI32) => {
```

Comment:
I think for now, we should comment out the debugging stuff (probably just big block comment) until our optimizations are a tad better.